discord-roles.md

```diff
@@ -0,0 +1,71 @@
+| Channel Manager        | PIM                               | Has Manage Channels permission                                    |        :x:         |        None         |                None                |
+| Announcer              | PIM                               | Has Write Message permission in #announcements/#dev-announcements |        :x:         |        None         |                None                |
+| Steering Council       | SC Members                        | Steering Council role, as we currently have it.                   | :white_check_mark: |        None         |                None                |
+| Moderators             | Moderators                        | Has moderation permissions, i.e. mute, kick, ban, etc.            | :white_check_mark: |        None         |                None                |
+| Maintainers            | Maintainers                       | Has GitHub permissions.                                           | :white_check_mark: |        None         |                None                |
+| Foundation Team        | Foundation Members                | Involved in the NeoForged legal entity.                           | :white_check_mark: |        None         |                None                |
+| Infrastructure Team    | Infrastructure Managers           | Involved in the project infrastructure in some way.               | :white_check_mark: |        None         |                None                |
+| Docs Team              | Documentation writers             | People with permissions on the docs repo.                         | :white_check_mark: |        None         |                None                |
+| Bot Team               | Bot managers                      | People who can manage the bot.                                    | :white_check_mark: |        None         |                None                |
+| NeoForged Team         | All team members                  | Catch-all for all the hoisted roles above.                        |        :x:         |        None         |                None                |
+| Automation             | Camelot                           | Contains camelot so that it can moderate the public.              |        :x:         |        Gray         |                Gear                |
+| Mojang Team            | Mojangsters                       | Mojang people.                                                    |        :x:         |     Mojang Red      |                None                |
+| ModLauncher Team       | McModLauncher Team                | Members of the ModLauncher side project.                          |        :x:         |        None         |                None                |
+| Moderator              | Moderators                        | The pingable moderator role.                                      |        :x:         |        None         |                None                |
```

Comment:
So, the reason we have a separate pingable role for Moderators in the server currently is because we have two 'classes' of Moderators:

- The full or regular moderators hold the `Moderator` role, with the yellow role name and yellow half-shield icon. These moderators are the full members of the team, with permissions to timeout, warn, kick, and ban as fits their duties.
- The moderators on probation, because of newly joining the team, hold the `Provisional Moderator` role, with the green role name and green outlined shield icon. These moderators have permission to timeout users, manage nicknames, messages, threads, but not the full powers given to Moderators. 
  The intention with this role is that these moderators have enough power for the small daily moderation tasks, but cannot do permanent actions (like kicks or bans) and must consult a regular moderator on the right course of action.

The `Moderators` pingable role is meant to _hoist_ both regular and provisional moderators in the same category under the members panel, since both classes of moderators are still meant to be approachable and pingable for moderation actions, but in a way that doesn't mean provisional moderators have full moderation permissions available to them (if we naively assigned them the Moderators role plus a probation marker role).

Not only is it meant to hoist both regular and provisional moderators on the same category, but also to hoist them separately from any other category, since I believe it to be a good principle that the community know who the moderators are, and who is currently online to be contacted (if not done via role mention).

Reply:
With respect to the probationary moderators rule. We discussed that in voice. Is that really necessary?

Practically speaking assuming somebody had ill-intentions then the probation will do jack shit, they can wait for three months. And then reak havoc.

It potentially guards against accidents, and that seems exactly not what the probationary moderator rule is designed for.

All in all, this role has practically no reason to exist, other as a safety guard for our own feelings when we add new moderators. Which is completely not needed, because there are many moderators, etc that already handle this. 

Practically speaking the Ban/Kick permission pair, should potentially not even be on moderators at all, but should be done through camelot, because that is a better system. So one could argue that the ban/kick role should be assigned to a special seperate role (Community Manager) and accessible through PIM.

Which is why the concept of seperate Probation Moderator as an entire role was removed, in favor of a general role "Probationary" that indicates that this user is a probie regardless of wether they are moderator or maintainer.

Reply:
I would disagree with ban/kick having to go through the bot. When someone is spamming nasty shit (eg. nsfw bots, which has happened), right-clicking on their name to ban and keeping the "delete messages from the last hour" option, is, in my opinion, much easier to do in a hurry or half-asleep than having to remember the specific sequence of commands needed to do the ban and cleanup manually.

I do however agree that provisional moderator isn't necessary. If we don't trust someone enough to give them a chance, they shouldn't be given a chance. A moderator, like maintainer, should already be known to the community and have proven themselves to be active and helpful in the community in a way that aligns with the project's community principles.

Reply:
I fully disagree with Giga's statement here, requiring bot auth every time a troll needs to be banned is too much. OTOH probation with separate perms is probably not required.

Reply:
I would need to check how the permission is handled. If we can grant the Discord based ban, then it should be easily possible to add it.

If that is not the case we can expand Camelot, because that can also offer you a ban option in the context menu.